packages/clerk-js/src/utils/web3.ts

```diff
@@ -1,46 +1,78 @@
 import type { Web3Provider } from '@clerk/shared/types';
+import type { SolanaWalletAdapterWallet } from '@solana/wallet-standard';
 
 import { clerkUnsupportedEnvironmentWarning } from '@/core/errors';
+import { getInjectedWeb3SolanaProviders } from '@/utils/injectedWeb3SolanaProviders';
 
 import { toHex } from './hex';
-import { getInjectedWeb3Providers } from './injectedWeb3Providers';
+import { getInjectedWeb3EthProviders } from './injectedWeb3EthProviders';
 
 type GetWeb3IdentifierParams = {
   provider: Web3Provider;
+  walletName?: string;
 };
 
+// '@solana/wallet-standard'
+const StandardConnect = `standard:connect`;
+const SolanaSignMessage = `solana:signMessage`;
+
 export async function getWeb3Identifier(params: GetWeb3IdentifierParams): Promise<string> {
-  const { provider } = params;
-  const ethereum = await getEthereumProvider(provider);
+  const { provider, walletName } = params;
+  const walletProvider = await getWeb3WalletProvider(provider, walletName);
 
   // TODO - core-3: Improve error handling for the case when the provider is not found
-  if (!ethereum) {
+  if (!walletProvider) {
     // If a plugin for the requested provider is not found,
     // the flow will fail as it has been the expected behavior so far.
     return '';
   }
 
-  const identifiers = await ethereum.request({ method: 'eth_requestAccounts' });
+  if (provider === 'solana') {
+    const identifiers = await walletProvider.features[StandardConnect].connect();
+    return (identifiers && identifiers.accounts[0].address) || '';
+  }
+
+  // Ethereum providers
+  const identifiers = await walletProvider.request({ method: 'eth_requestAccounts' });
   // @ts-ignore -- Provider SDKs may return unknown shape; use first address if present
   return (identifiers && identifiers[0]) || '';
 }
 
 type GenerateWeb3SignatureParams = GenerateSignatureParams & {
   provider: Web3Provider;
+  walletName?: string;
 };
 
 export async function generateWeb3Signature(params: GenerateWeb3SignatureParams): Promise<string> {
   const { identifier, nonce, provider } = params;
-  const ethereum = await getEthereumProvider(provider);
+  const wallet = await getWeb3WalletProvider(provider, params?.walletName);
```

Comment:
let's align our naming here across the methods, we can choose either walletProvider or just wallet